### PR TITLE
MB-601: update shipments endpoint estimated weight

### PR DESCRIPTION
**Pull request by ralren**
## Description

Update the shipments endpoint to give the Prime ability to send estimated weight. The date this is sent will be recorded. If there is already weight recorded, the request will auto fail. There are also other cases to consider:

* If the date the shipment was approved was at least ten days from the scheduled move date, the estimated weight must have been recorded at least ten days from scheduled move date as well.

* If the shipment was approved between 3 to 9 days from the scheduled move date, the estimated weight must have been recorded at least three days from the scheduled move date.

* Finally, if the shipment was approved less than three days from the scheduled move date, the estimated wight must have been recorded at least one day from the scheduled move date.

As there was no previous record of when a shipment was approved, I added a new column to the `mto_shipments` date to capture this. This involved tweaking the `mto_shipment_status_updater`.

## Setup

```sh
make server_run
```

Use Postman or run tests.

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-601) for this change

### CODE REVIEW

**Comment by LeDeep** [comment]
not sure what happened, but the last commit deleted all the `pkg/gen` files.

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 1]
```diff
@@ -123,52 +136,109 @@ func validateUpdatedMTOShipment(db *pop.Connection, oldShipment *models.MTOShipm
...
 		return err
 	}
 	return nil
 }
 
+func validatePrimeEstimatedWeightRecordedDate(estimatedWeightRecordedDate time.Time, scheduledPickupDate time.Time, approvedDate time.Time) error {
+	approvedDaysFromScheduled := scheduledPickupDate.Sub(approvedDate).Hours() / 24
+	daysFromScheduled := scheduledPickupDate.Sub(estimatedWeightRecordedDate).Hours() / 24
+	if approvedDaysFromScheduled >= 10 && daysFromScheduled >= 10 {
+		return nil
```
Comment (LeDeep) [inline pkg/services/mto_shipment/mto_shipment_updater.go]: can the Prime return an estimated weight before the TOO approves?

Comment (ralren) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: let's ask product!

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 2]
```diff
@@ -159,27 +198,34 @@ func updateMTOShipment(db *pop.Connection, mtoShipmentID uuid.UUID, unmodifiedSi
 		updatedShipment.DestinationAddress.ID,
 	)
 
+	if updatedShipment.PrimeEstimatedWeight != 0 {
+		estimatedWeightQuery := `,
+			prime_estimated_weight = %d,
+			prime_estimated_weight_recorded_date = NOW()`
+		baseQuery = baseQuery + fmt.Sprintf(estimatedWeightQuery, updatedShipment.PrimeEstimatedWeight)
```
Comment (ralren) [inline pkg/services/mto_shipment/mto_shipment_updater.go]: forgot to refactor this!

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 1 above)
Comment (LeDeep) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: looks like shipment will have been approved by this point. https://trussworks.slack.com/archives/CC6UEBENR/p1582056295486600

File: migrations/app/secure/20200211184638_add_approved_date_to_mto_shipments.up.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE mto_shipments
+	ADD COLUMN approved_date date;
```
Comment (chrisgilmerproj) [inline migrations/app/secure/20200211184638_add_approved_date_to_mto_shipments.up.sql]: It looks like you allow this to be null below with `*time.Time`. Did you want to allow null here?

Comment (ralren) [reply migrations/app/secure/20200211184638_add_approved_date_to_mto_shipments.up.sql]: Yes sir.

Comment (chrisgilmerproj) [reply migrations/app/secure/20200211184638_add_approved_date_to_mto_shipments.up.sql]: I mean to ask, is `date` type here nullable by default? Or were you supposed to have marked it somehow? I'm reading the docs and it appears that this defaults to be a nullable column unless you say `NOT NULL`. So I think I didn't have any reason to be worried here. But I wanted to check with you.

https://www.postgresql.org/docs/10/sql-altertable.html

**Pull request merged**